torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -224,6 +241,25 @@ void RRefContext::delAllUsers(std::chrono::milliseconds timeoutMillis) {
     rref_ptr->tryDel();
   }
 
+  // If an rref in the owners_ map has never been forked, we will never get a
+  // corresponding message from the forking node(s) telling us to delete the
+  // RRef. Hence we delete the RRef here. This can occur when a remote call is
+  // sent to self and times out.
```

Comment:
@mrshenli , since the shutdown sequence is currently:
_delete_all_user_rrefs() (which calls this function)
_destroy_rref_context() (which also clears the owners map)

Since we run `delAllUsers` first, this deletion is here to avoid the long CV-based wait that waits for the owners map to be cleared. If we didn't have this and had the wait, then unittests could be flaky and time out (since if the condition described in the comment happens, owners_ size will be > 0).

Reply:
I see. Thanks for explaining.